### PR TITLE
Priy1289/dictionaryrenderergraphicsoverlay

### PR DESCRIPTION
**Pull request by priy1289**
sample for dictionary renderer with graphics overlay

### CODE REVIEW

File: java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class MainActivity extends AppCompatActivity {
+
+  // permission to read external storage
+  private final String[] reqPermission = new String[] { Manifest.permission.READ_EXTERNAL_STORAGE };
+
+  private MapView mMapView;
+
```
Comment (TADraeseke) [inline java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java]: remove white space

File: java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+  private GraphicsOverlay graphicsOverlay;
```
Comment (TADraeseke) [inline java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java]: rename with standard naming convention for fields, mGraphicsOverlay

File: java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    // get the reference to the map view
+    mMapView = findViewById(R.id.mapView);
+
```
Comment (TADraeseke) [inline java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java]: remove white space

File: java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+      applyDictionaRendererToGraphics();
+    } else {
+      // request permission
+      int requestCode = 2;
+      ActivityCompat.requestPermissions(this, reqPermission, requestCode);
+    }
+  }
+
+  @TargetApi(24)
+  private void applyDictionaRendererToGraphics() {
```
Comment (TADraeseke) [inline java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java]: mispelling in method name

File: java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+    if (ContextCompat.checkSelfPermission(this, reqPermission[0]) == PackageManager.PERMISSION_GRANTED) {
+      applyDictionaRendererToGraphics();
+    } else {
+      // request permission
+      int requestCode = 2;
+      ActivityCompat.requestPermissions(this, reqPermission, requestCode);
+    }
+  }
+
+  @TargetApi(24)
```
Comment (TADraeseke) [inline java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java]: All our apps need to run back to API 19. Please remove

File: java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+    // tells graphics overlay how to render graphics with symbol dictionary attributes set
+    DictionaryRenderer renderer = new DictionaryRenderer(symbolDictionary);
+    graphicsOverlay.setRenderer(renderer);
+
+    // parse graphic attributes from a XML file
+    List<Map<String, Object>> messages = parseMessages();
+
+    // create graphics with attributes and add to graphics overlay
+    messages.stream()
+        .map(MainActivity::createGraphic)
```
Comment (TADraeseke) [inline java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java]: Can't use streams before api 24

File: java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+        .collect(Collectors.toCollection(() -> graphicsOverlay.getGraphics()));
+
+    // once view has loaded
+    mMapView.addSpatialReferenceChangedListener(e -> {
```
Comment (TADraeseke) [inline java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java]: do we want to listen for spatial reference change for this? or for graphics overlay to have loaded?

File: java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+      // set initial viewpoint
+      mMapView.setViewpointGeometryAsync(graphicsOverlay.getExtent());
+    });
+  }
+
+  /**
+   * Parses a XML file and creates a message for each block of attributes found.
+   */
+  private List<Map<String, Object>> parseMessages() {
```
Comment (TADraeseke) [inline java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java]: This method could benefit from some in line comments to say what each part is doing

File: java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+              }
+            }
+            break;
+        }
+
+        eventType = parser.next();
+      }
+
+    } catch (XmlPullParserException e) {
+
```
Comment (TADraeseke) [inline java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java]: None of these errors are handled

File: java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+    return messages;
+  }
+
+  /**
+   * Creates a graphic using a symbol dictionary and the attributes that were passed.
+   *
+   * @param attributes tells symbol dictionary what symbol to apply to graphic
+   */
+  @TargetApi(24)
```
Comment (TADraeseke) [inline java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java]: Can't target api 24

File: java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java [hunk 11]
```diff
@@ -0,0 +1,235 @@
...
+import android.support.annotation.NonNull;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+
+import android.widget.Toast;
+import com.esri.arcgisruntime.geometry.Multipoint;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.geometry.PointCollection;
```
Comment (TADraeseke) [inline java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java]: wrong order and spacing on imports

File: java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java [hunk 12]
```diff
@@ -0,0 +1,235 @@
...
+  private static Graphic createGraphic(Map<String, Object> attributes) {
+
+    // get spatial reference
+    int wkid = Integer.parseInt((String) attributes.get("_wkid"));
+    SpatialReference sr = SpatialReference.create(wkid);
+
+    // get points from coordinates' string
+    PointCollection points = new PointCollection(sr);
+    String[] coordinates = ((String) attributes.get("_control_points")).split(";");
+    Stream.of(coordinates)
```
Comment (TADraeseke) [inline java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java]: Can't use streams on phones before API 24

File: java/dictionary-renderer-graphics-overlay/README.md [hunk 13]
```diff
@@ -0,0 +1,38 @@
+# Dictionary Renderer with Graphics Overlay
+
+Display mil2525d symbols.
```
Comment (TADraeseke) [inline java/dictionary-renderer-graphics-overlay/README.md]: Lets just make it:

~Display mil2525d symbols.~
~ The dictionary renderer~ Creates ~these~ graphics using a local mil2525d style file and a XML file with key, value attributes for each graphic.

File: java/dictionary-renderer-graphics-overlay/README.md [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+
+The dictionary renderer creates these graphics using a local mil2525d style file and a XML file with key, value attributes for each graphic.
+
+![Dictionary Renderer Graphics Overlay App](dictionary-renderer-graphics-overlay.png)
+
+## How it works
+
+This sample loads a number of point, line, and polygon military elements from an XML file and adds them as graphics to a `GraphicsOverlay`. A `DictionaryRenderer` is applied to the `GraphicsOverlay` in order to display the graphics with MIL-STD-2525D military symbology. When all graphics are created, the map's viewpoint is set to zoom to the full extent of all graphics.
```
Comment (TADraeseke) [inline java/dictionary-renderer-graphics-overlay/README.md]: can drop the line about setting the map's viewpoint

File: java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java (hunk 11 above)
Comment (priy1289) [reply java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java]: corrected the spacing. 
Didn't get what is wrong with the order?

File: java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java (hunk 8 above)
Comment (priy1289) [reply java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java]: done. Added comments

File: java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java (hunk 9 above)
Comment (priy1289) [reply java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java]: Sorry, missed it. Handled now.

File: java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java (hunk 7 above)
Comment (priy1289) [reply java/dictionary-renderer-graphics-overlay/src/main/java/com/esri/arcgisruntime/sample/dictionaryrenderergraphicsoverlay/MainActivity.java]: I couldn't find a listener for when graphic overlay gets loaded, so I can call that listener here.

**Pull request merged**